src/ui/zcl_abapgit_gui_chunk_lib.clas.abap

```diff
@@ -785,7 +785,10 @@ CLASS zcl_abapgit_gui_chunk_lib IMPLEMENTATION.
           lv_display_url       TYPE zif_abapgit_persistence=>ty_repo-url,
           lv_icon_commit       TYPE string.
 
-    lv_commit_hash = io_repo_online->get_current_remote( ).
+    lv_commit_hash = io_repo_online->get_selected_commit( ).
+    IF lv_commit_hash IS INITIAL.
+      lv_commit_hash = io_repo_online->get_current_remote( ).
+    ENDIF.
```

Comment:
```suggestion
    lv_commit_hash = io_repo_online->get_current_remote( ).
```

Reply:
> `get_current_remote` will get the current remote? and take care of it to be correct, for branches, for tags, for selected commits? to get_current_remote it should not matter from the outside, what has been selected by the user?

Ahh, yes, we differentiate between them already in fetch_remote..